conan/tools/gnu/autotoolstoolchain.py

```diff
@@ -109,12 +109,24 @@ def _resolve_android_cross_compilation(self):
         ret = {}
         if not self._is_cross_building or not self._conanfile.settings.get_safe("os") == "Android":
             return ret
+        # User variables have more priority than Conan ones
+        build_env = VirtualBuildEnv(self._conanfile, auto_generate=True).vars()
+        for var in ["CC", "CXX", "LD", "STRIP", "RANLIB", "AS", "AR", "host"]:
```

Comment:
May I suggest also including `["ADDR2LINE", "NM", "OBJCOPY", "OBJDUMP", "READELF", "ELFEDIT"]` from here: https://github.com/conan-io/conan-center-index/blob/4398cb26c62f38a6822eb32fb0eb61af5b0f665f/recipes/android-ndk/all/conanfile.py#L292-L315 ?
Having cross-compiled nearly 100% of the packages on CCI, the most common additional tools to be provided as env vars have been ar, strip, as, ranlib, objdump, and elfedit. In that order, the latter two only being required by `perf`.

Reply:
Thanks for the suggestion @valgur I think it makes sense 👍 
I just added them.